satellite/orders/endpoint.go

```diff
@@ -167,55 +189,31 @@ func (endpoint *Endpoint) Settlement(stream pb.Orders_SettlementServer) (err err
 			continue
 		}
 
-		bucketID, err := endpoint.DB.UseSerialNumber(ctx, orderLimit.SerialNumber, orderLimit.StorageNodeId)
-		if err != nil {
-			log.Warn("unable to use serial number", zap.Error(err))
-			if ErrUsingSerialNumber.Has(err) {
-				err := monitoredSettlementStreamSend(ctx, stream, &pb.SettlementResponse{
-					SerialNumber: orderLimit.SerialNumber,
-					Status:       pb.SettlementResponse_REJECTED,
-				})
-				if err != nil {
-					return formatError(err)
-				}
-				continue
-			}
-			return err
-		}
-		now := time.Now().UTC()
-		intervalStart := time.Date(now.Year(), now.Month(), now.Day(), now.Hour(), 0, 0, 0, now.Location())
+		requests = append(requests, &ProcessOrderRequest{Order: order, OrderLimit: orderLimit})
 
-		projectID, bucketName, err := SplitBucketID(bucketID)
-		if err != nil {
-			return err
-		}
-		if err := endpoint.DB.UpdateBucketBandwidthSettle(ctx, *projectID, bucketName, orderLimit.Action, order.Amount, intervalStart); err != nil {
-			// TODO: we should use the serial number in the same transaction we settle the bandwidth? that way we don't need this undo in an error case?
-			if err := endpoint.DB.UnuseSerialNumber(ctx, orderLimit.SerialNumber, orderLimit.StorageNodeId); err != nil {
-				log.Error("unable to unuse serial number", zap.Error(err))
+		if len(requests) == endpoint.settlementBatchSize {
```

Comment:
This may seem overly paranoid, but I'd prefer we change to `>=` instead of `==`, just in case something weird occurs in the future where we skip the exact batch size.

Reply:
Updated